package/kernel/linux/modules/netdevices.mk

```diff
@@ -407,6 +407,40 @@ endef
 
 $(eval $(call KernelPackage,phy-aquantia))
 
+
+define KernelPackage/switch-b53
+  SUBMENU:=$(NETWORK_DEVICES_MENU)
+  TITLE:=Broadcom BCM53xx managed switch DSA support
+  KCONFIG:=CONFIG_B53 \
+  CONFIG_NET_DSA_TAG_BRCM=y \
+  CONFIG_NET_DSA=y
```

Comment:
This will activate `CONFIG_NET_DSA` in all kernel images of all targets in the buildbot builds. 
The `kmod-dsa-mv88e6xxx` does it in a similar way. 

`CONFIG_NET_DSA` is a tristate option we should add a `kmod-dsa` to build the DSA subsystem as a module for the boards which do not have DSA activated.

Reply:
I would do it like this:
1. new `kmod-dsa` which builds `CONFIG_NET_DSA` and packages the `net/dsa/dsa_core.ko`
2. Make `kmod-dsa-mv88e6xxx` and `kmod-dsa-tag-dsa` depend on `kmod-dsa` and remove `CONFIG_NET_DSA=y` from their kmod.
3. Rename your `kmod-switch-b53` to `kmod-dsa-b53` which contains `CONFIG_B53`, `CONFIG_NET_DSA_TAG_BRCM`,  `CONFIG_NET_DSA_TAG_BRCM_LEGACY` and `CONFIG_NET_DSA_TAG_BRCM_PREPEND` and the 4 ko files generated by those options.

Reply:
Thanks for your tips, @hauke.

In the end, this made the switch work without adding CONFIG_NET_DSA as a built-in module:

```
define KernelPackage/dsa-tag-brcm
  SUBMENU:=$(NETWORK_DEVICES_MENU)
  TITLE:=Broadcom DSA tagging support
  DEPENDS:=+kmod-dsa
  KCONFIG:= CONFIG_NET_DSA_TAG_BRCM_COMMON
  FILES:=$(LINUX_DIR)/net/dsa/tag_brcm.ko
  AUTOLOAD:=$(call AutoLoad,40,tag_brcm,1)
endef

define KernelPackage/dsa-tag-brcm/description
  Kernel module for Broadcom DSA tagging
endef

$(eval $(call KernelPackage,dsa-tag-brcm))


define KernelPackage/dsa-b53
  SUBMENU:=$(NETWORK_DEVICES_MENU)
  TITLE:=Broadcom BCM53xx managed switch DSA support
  DEPENDS:=+kmod-dsa-tag-brcm
  KCONFIG:=CONFIG_B53 \
  CONFIG_NET_DSA_TAG_BRCM=y
  FILES:=$(LINUX_DIR)/drivers/net/dsa/b53/b53_common.ko
  AUTOLOAD:=$(call AutoProbe,b53_common)
endef

define KernelPackage/dsa-b53/description
  Broadcom BCM53xx managed switch support
endef

$(eval $(call KernelPackage,dsa-b53))


define KernelPackage/dsa-b53-mdio
  SUBMENU:=$(NETWORK_DEVICES_MENU)
  TITLE:=B53 MDIO connected switch DSA driver
  DEPENDS:=+kmod-dsa-b53
  KCONFIG:=CONFIG_B53_MDIO_DRIVER
  FILES:=$(LINUX_DIR)/drivers/net/dsa/b53/b53_mdio.ko
  AUTOLOAD:=$(call AutoProbe,b53_mdio)
endef

define KernelPackage/dsa-b53-mdio/description
  B53 MDIO connected switch driver
endef

$(eval $(call KernelPackage,dsa-b53-mdio))
```

Nevertheless, now I have a working setting, I'll give another try to your suggestion.

Reply:
It looks like the CI tests report problems with the old b53 driver. May I add the CONFLICTS? Like: 

```
  define KernelPackage/dsa-b53
  SUBMENU:=$(NETWORK_DEVICES_MENU)
  TITLE:=Broadcom BCM53xx managed switch DSA support
  CONFLICTS:=kmod-switch-bcm53xx
  DEPENDS:=+kmod-dsa
  KCONFIG:=CONFIG_B53 \
  CONFIG_NET_DSA_TAG_BRCM \
  CONFIG_NET_DSA_TAG_BRCM_LEGACY \
  CONFIG_NET_DSA_TAG_BRCM_PREPEND
  FILES:= \
  $(LINUX_DIR)/drivers/net/dsa/b53/b53_common.ko \
  $(LINUX_DIR)/net/dsa/tag_brcm.ko
  AUTOLOAD:=$(call AutoProbe,b53_common)
endef

define KernelPackage/dsa-b53/description
  Broadcom BCM53xx managed switch support
endef

$(eval $(call KernelPackage,dsa-b53))


define KernelPackage/dsa-b53-mdio
  SUBMENU:=$(NETWORK_DEVICES_MENU)
  TITLE:=B53 MDIO connected switch DSA driver
  CONFLICTS:=kmod-switch-bcm53xx-mdio
  DEPENDS:=+kmod-dsa-b53
  KCONFIG:=CONFIG_B53_MDIO_DRIVER
  FILES:=$(LINUX_DIR)/drivers/net/dsa/b53/b53_mdio.ko
  AUTOLOAD:=$(call AutoProbe,b53_mdio)
endef
```